digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/NodesAndJunctionsService.scala

```diff
@@ -338,6 +339,10 @@ class NodesAndJunctionsService(roadwayDAO: RoadwayDAO, roadwayPointDAO: RoadwayP
         }
         logger.info(s"Creating JunctionPoint with roadwayNumber : ${r.roadwayNumber} addrM: ${r.endAddrMValue} beforeAfter: ${BeforeAfter.Before.value}, junctionId: $junctionId")
         junctionPointDAO.create(Seq(JunctionPoint(NewIdValue, BeforeAfter.Before, rwPoint, junctionId, None, None, DateTime.now, None, username, Some(DateTime.now), r.roadwayNumber, r.endAddrMValue, r.roadNumber, r.roadPartNumber, r.track, r.discontinuity))).head
+        if (CalibrationPointDAO.fetch(r.linkId, CalibrationPointLocation.EndOfLink, rwPoint).isEmpty) {
```

Comment:
This code block is duplicated eight times so might be good idea to create one method that takes linkId, CalibrationPointLocation, rwPoint and junctionId as a parameter.

Reply:
I'm working on this...